how-to/integrate-with-sso/integrate-with-okta/client/src/auth.ts

```diff
@@ -0,0 +1,156 @@
+import { OktaAuth } from "@okta/okta-auth-js";
+import { getAccessToken } from "./app";
+import { getSettings } from "./settings";
+import type { OktaSettings } from "./shapes";
+
+let authSettings: OktaSettings;
+let informationCallback: (info: string) => void;
+
+/*
+code_verifier: cryptographically random string using the chars A-Z, a-z, 0-9 and the punctuation chars hyphen, period, underscore and tilde.
+codeChallenge: should be a calculated value of Base64-URL-encoded SHA256 hash of code_verifier.
+*/
+const codeVerifier = "Zg6klgrnixQJ629GsawRMV8MjWvwRAr-vyvP1MHnB6X8WKZN";
+const codeChallenge = "iF_7prUeJ6rr3jMG3LmhW3R1cZ2ecZavFqS0jtb6tzo";
+
+export async function init(settings: OktaSettings, informationCb: (info: string) => void) {
+	informationCallback = informationCb;
+
+	informationCallback("Initialising the authentication");
+
+	authSettings = settings;
+	if (!authSettings) {
+		informationCallback("Error: Settings missing cannot continue");
+	}
+}
+
+function makeid(length): string {
+	let result = "";
+	const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-";
+	const charactersLength = characters.length;
+	let counter = 0;
+	while (counter < length) {
+		result += characters.charAt(Math.floor(Math.random() * charactersLength));
+		counter += 1;
+	}
+	return result;
+}
+
+// PKCE workflow - https://developer.okta.com/blog/2019/08/22/okta-authjs-pkce
+export async function login() {
+	const sessionSettings = await getSettings();
+	const state = makeid(32);
+
+	const authUrl =
+		`${sessionSettings.okta.domain}v1/authorize?client_id=${sessionSettings.okta.clientId}&scope=openid email phone address groups profile` +
+		"&response_type=code" +
+		"&response_mode=query" +
+		"&nonce=nonceStatic" +
+		`&redirect_uri=${sessionSettings.okta.redirectUrl}` +
+		`&code_challenge=${codeChallenge}
+		&code_challenge_method=S256` +
+		`&state=${state}&sessionToken=session_not_needed`;
+
+	await showWindow(authUrl);
```

Comment:
This flow is still not correct, once the window is shown the code should be polling for the window to redirect to the login.html url. It should then grab the access code from the url. Then close the login window html and open the app html. This is the flow used in Auth0 example, so that should be good as the basis

Reply:
I was going by Okta's documentation - https://developer.okta.com/docs/guides/sign-into-spa-redirect/angular/main/#set-up-okta. I will look into how you have implemented in Auth0 and attempt it that way.